sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -150,9 +439,20 @@ public static Object convertAvroFieldStrict(
       @Nonnull org.apache.avro.Schema avroSchema,
       @Nonnull Schema.FieldType fieldType) {
 
-    org.apache.avro.Schema unwrapped = unwrapNullableSchema(avroSchema);
+    TypeWithNullability type = new TypeWithNullability(avroSchema);
+    LogicalType logicalType = LogicalTypes.fromSchema(type.type);
+    if (logicalType != null) {
+      if (logicalType instanceof LogicalTypes.Decimal) {
+        BigDecimal bigDecimal =
+            new Conversions.DecimalConversion()
+                .fromBytes((ByteBuffer) value, type.type, logicalType);
```

Comment:
As I understand, we need to do `.duplicate()` for the same reason as in `convertBytesStrict`

Reply:
good catch here!